### PR TITLE
Document the signing API (bug 1209230)

### PR DESCRIPTION
**Pull request by mstriemer**
r? @andymckay 

Does this seem right for API docs? On Marketplace we had a pretty rigid format. Should I set something like that up and move this into that format or is that a task for later?

### CODE REVIEW

**Comment by andymckay** [comment]
We have sphinxhttp in the addons docs already. But I'm keen to get docs, and we can re-format once they land.

File: docs/topics/api/signing.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+Once validation is completed (whether it passes or fails) then the
+``processed`` property will be ``true``. You can check if validation
+passed using the ``valid`` property and check the results with
+``validation_results``.
+
+If validation passed then your add-on will be submitted for review.
+In the case of unlisted add-ons this will happen automatically if
+the add-on passes a strict set of tests. If your add-on is listed
+then it will be reviewed by a human and that will take a bit
+longer. Once review is complete then the ``reviewed`` property
```
Comment (andymckay) [inline docs/topics/api/signing.rst]: Is listed something we are doing right now? I guess you can upload a new version of a listed add-on through this API?

**Comment by andymckay** [comment]
r+wc, I'll give it a reformat for you :)

File: docs/topics/api/signing.rst (hunk 1 above)
Comment (mstriemer) [inline docs/topics/api/signing.rst]: Yeah, if you already have a listed add-on then it will accept a new version and go into the regular review queue.

**Pull request merged**